packages/advanced-logic/src/extensions/payment-network/erc777/stream.ts

```diff
@@ -1,25 +1,23 @@
 import { ExtensionTypes, RequestLogicTypes, TypesUtils } from '@requestnetwork/types';
 import ReferenceBasedPaymentNetwork from '../reference-based';
 import Utils from '@requestnetwork/utils';
+import EvmBasedPaymentNetwork from '../evm-based';
 const CURRENT_VERSION = '0.1.0';
 
 /**
  * Implementation of the payment network to pay in ERC777, including third-party fees payment, based on a reference provided to a proxy contract.
  */
 export default class Erc777StreamPaymentNetwork<
-  TCreationParameters extends ExtensionTypes.PnStreamReferenceBased.ICreationParameters = ExtensionTypes.PnStreamReferenceBased.ICreationParameters,
-> extends ReferenceBasedPaymentNetwork<TCreationParameters> {
+    TCreationParameters extends ExtensionTypes.PnStreamReferenceBased.ICreationParameters = ExtensionTypes.PnStreamReferenceBased.ICreationParameters,
+  >
+  extends ReferenceBasedPaymentNetwork<TCreationParameters>
+  implements EvmBasedPaymentNetwork
+{
   public constructor(
-    extensionId: ExtensionTypes.ID = ExtensionTypes.ID.PAYMENT_NETWORK_ERC777_STREAM,
-    currentVersion: string = CURRENT_VERSION,
-    public supportedNetworks: string[] = [
-      'matic',
-      'xdai',
-      'mumbai',
-      'rinkeby',
-      'goerli',
-      'arbitrum-rinkeby',
-    ],
+    public readonly extensionId: ExtensionTypes.ID = ExtensionTypes.ID
+      .PAYMENT_NETWORK_ERC777_STREAM,
+    public readonly currentVersion: string = CURRENT_VERSION,
+    public readonly supportedNetworks: string[] = EvmBasedPaymentNetwork.EVM_NETWORKS,
```

Comment:
@alexandre-abrioux  ERC777 is not supported on all networks, most importantly mainnet is not supported

Reply:
@KolevDarko I took a different approach in the end as advised by Benji. I'm working on it here: https://github.com/RequestNetwork/requestNetwork/pull/950